src/qililab/platform/platform.py

```diff
@@ -528,15 +529,17 @@ def __str__(self) -> str:
         """
         return str(YAML().dump(self.to_dict(), io.BytesIO()))
 
-    def execute_qprogram(self, qprogram: QProgram):
+    def execute_qprogram(self, qprogram: QProgram) -> dict[str, list[Result]]:
```

Comment:
Before the official release of `QProgram` (which should be December 1st) can we have a `Results` class instead of a dictionary?

Reply:
When doing measurements on different qubits this class will be very helpful to get the probabilities of all the qubits with just one call

Reply:
Yes we can! I didn't want to add it in this PR because it is already huge.

Reply:
Completely agree!